src/controllers/RideOffer.js

```diff
@@ -1,52 +1,125 @@
 import Model from '../models/model';
 
-const driverOfferModel = new Model('ride_offer');
+const offerModel = new Model('ride_offer');
+/**
+ * @description - This method add offer to the database
+ *
+ * @param {object} req - request
+ *
+ * @param {object} res - response
+ *
+ * @return {object} - it return data object
+ */
 export const addOffer = async (req, res) => {
-  const {
-    driverId, destination, price
-  } = req.body;
-  const columns = 'driver_id, destination, price';
-  const values = `'${driverId}', '${destination}', '${price}'`;
+  const { data: { id } } = req.user;
+  const { destination, price } = req.body;
+  const columns = '"driverId", destination, price';
+  const values = `'${id}', '${destination}', '${price}'`;
   try {
-    const data = await driverOfferModel.insertWithReturn(columns, values);
-    res.status(200).json({ msg: data.rows });
+    const data = await offerModel.insertWithReturn(columns, values);
+    return res.status(201)
+      .json({ data: data.rows[0], message: 'offer created successfully', success: true });
   } catch (err) {
-    res.status(400).json({ msg: err.stack });
+    res.status(500).json({ message: err.message });
   }
 };
-
+/**
+ * @description - This method will return a offer for particular driver
+ *
+ * @param {object} req - request
+ *
+ * @param {object} res - response
+ *
+ * @return {object} - it return object
+ */
 export const DriverRideOfferPage = async (req, res) => {
-  const { id } = req.user.driver;
+  const { data: { id } } = req.user;
   try {
-    const data = await driverOfferModel.select('*', `WHERE "driver_id" = '${id}'`);
-    res.status(200).json({ msg: data.rows });
+    const data = await offerModel.select('*', `WHERE "driverId" = '${id}'`);
+    if (data.rowCount === 0) {
+      return res.status(404).json(
+        { data: [], message: 'data does not exist', success: false }
+      );
+    }
+    return res.status(201).json({
+      data: data.rows, success: true
+    });
   } catch (err) {
-    res.status(400).json({ msg: err.stack });
+    res.status(500).json({ message: err.message });
   }
 };
+/**
+ *@description - This method fetch and return all offer object
 
-export const modifyOfferpage = async (req, res) => {
-  const { destination, price } = req.body;
+ * @param {object} req - request
+ *
+ * @param {object} res - response
+ *
+ * @returns {object} - it return a data object and the number of the data
+ */
+export const allOffer = async (req, res) => {
+  let total;
+  try {
+    const data = await offerModel.select('*');
+    total += data.rowCount;
+    if (!data.rowCount) {
+      return res.status(404).json(
+        { data: [], message: 'data does not exist', success: false }
+      );
+    }
+    return res.status(201).json({ data: data.rows, total, success: true, });
```

Comment:
Also `200` would be ideal instead of `201` - read more about http status code here
https://en.wikipedia.org/wiki/List_of_HTTP_status_codes